### PR TITLE
Issue 495 ASN support

### PR DESCRIPTION
**Pull request by LuitelSamikshya**
This PR adds ASN support.

```console
chaos -d tesla.com | ./httpx -asn

    __    __  __       _  __
   / /_  / /_/ /_____ | |/ /
  / __ \/ __/ __/ __ \|   /
 / / / / /_/ /_/ /_/ /   |
/_/ /_/\__/\__/ .___/_/|_|
             /_/              v1.2.1-dev

		projectdiscovery.io

Use with caution. You are responsible for your actions.
Developers assume no liability and are not responsible for any misuse or damage.
http://autodiscover.tesla.com [AS8075, MICROSOFT-CORP-MSN-AS-BLOCK, US, 40.96.0.0/13]
https://akamai-apigateway-profileapi.tesla.com [AS16625, AKAMAI-AS, US, 184.29.74.0/23]
https://akamai-apigateway-vehicleextinfogw-prdsvc-st.tesla.com [AS16625, AKAMAI-AS, US, 184.29.74.0/23]
https://auth-global.tesla.com [AS16625, AKAMAI-AS, US, 184.29.74.0/23]
https://cdns-content.tesla.com [AS16625, AKAMAI-AS, US, 184.29.74.0/23]
https://auth.tesla.com [AS16625, AKAMAI-AS, US, 104.122.0.0/20]
https://apacvpn.tesla.com [AS394161, TESLA, US, 8.244.67.0/24]
https://auth-global-stage.tesla.com [AS16625, AKAMAI-AS, US, 104.122.0.0/20]
https://cdn-design.tesla.com [AS16625, AKAMAI-AS, US, 104.122.0.0/20]
https://digitalassets-energy.tesla.com [AS54113, FASTLY, US, 151.101.0.0/22]
https://apps.tesla.com [AS16625, AKAMAI-AS, US, 184.29.74.0/23]
https://business-ui-ownership.tesla.com [AS16625, AKAMAI-AS, US, 104.122.0.0/20]
https://3.tesla.com [AS16625, AKAMAI-AS, US, 104.122.0.0/20]
```

### CODE REVIEW

File: runner/runner.go [hunk 1]
```diff
@@ -927,6 +928,25 @@ retry:
 		fullURL = parsedURL.String()
 	}
 
+	var asnResponse interface{ String() string }
+	if r.options.Asn {
+		addr, err := net.LookupIP(domain)
```
Comment (Mzack9999) [inline runner/runner.go:933]: We can skip this lookup by moving the logic later and reusing the `ip` variable value at https://github.com/projectdiscovery/httpx/blob/1ca3fb668c64f0b3a0016b5850647ed44c9f3bb1/runner/runner.go#L1130

File: runner/runner.go [hunk 2]
```diff
@@ -927,6 +928,25 @@ retry:
...
+		if err != nil {
+			return Result{URL: URL.String(), Input: origInput, err: err}
+		}
+		lookupResult, err := ipisp.LookupIP(context.Background(), addr[0])
+		if err != nil {
+			return Result{URL: URL.String(), Input: origInput, err: err}
```
Comment (Mzack9999) [inline runner/runner.go:939]: The failed lookup shouldn't be fatal, but print an error warning and leave the variable to nil

**Review by Mzack9999 — CHANGES_REQUESTED** [review]

**Review by ehsandeep — CHANGES_REQUESTED** [review]
```console
chaos -d tesla.com | ./httpx -asn

    __    __  __       _  __
   / /_  / /_/ /_____ | |/ /
  / __ \/ __/ __/ __ \|   /
 / / / / /_/ /_/ /_/ /   |
/_/ /_/\__/\__/ .___/_/|_|
             /_/              v1.2.1-dev

		projectdiscovery.io

Use with caution. You are responsible for your actions.
Developers assume no liability and are not responsible for any misuse or damage.
http://autodiscover.tesla.com [AS8075, MICROSOFT-CORP-MSN-AS-BLOCK, US, 40.96.0.0/13]
https://akamai-apigateway-profileapi.tesla.com [AS16625, AKAMAI-AS, US, 184.29.74.0/23]
https://akamai-apigateway-vehicleextinfogw-prdsvc-st.tesla.com [AS16625, AKAMAI-AS, US, 184.29.74.0/23]
https://auth-global.tesla.com [AS16625, AKAMAI-AS, US, 184.29.74.0/23]
https://cdns-content.tesla.com [AS16625, AKAMAI-AS, US, 184.29.74.0/23]
https://auth.tesla.com [AS16625, AKAMAI-AS, US, 104.122.0.0/20]
https://apacvpn.tesla.com [AS394161, TESLA, US, 8.244.67.0/24]
https://auth-global-stage.tesla.com [AS16625, AKAMAI-AS, US, 104.122.0.0/20]
https://cdn-design.tesla.com [AS16625, AKAMAI-AS, US, 104.122.0.0/20]
https://digitalassets-energy.tesla.com [AS54113, FASTLY, US, 151.101.0.0/22]
https://apps.tesla.com [AS16625, AKAMAI-AS, US, 184.29.74.0/23]
https://business-ui-ownership.tesla.com [AS16625, AKAMAI-AS, US, 104.122.0.0/20]
https://3.tesla.com [AS16625, AKAMAI-AS, US, 104.122.0.0/20]
[ERR] parse ASN "4812 17621": parse "4812 17621": strconv.Atoi: parsing "4812 17621": invalid syntax
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x102b1826c]

goroutine 396 [running]:
github.com/projectdiscovery/httpx/runner.(*Runner).analyze(0x1400000d560, 0x1400007d270, {0x102b2c7a4, 0xa}, {0x140009ad0e0, 0xf}, {0x102b1f083, 0x3}, {0x140009ad0e0, 0xf}, ...)
	/Users/geekboy/Github/httpx/runner/runner.go:1138 +0x3fcc
github.com/projectdiscovery/httpx/runner.(*Runner).process.func1(0x14000156940, 0x1400000d560, 0x1400007d270, {0x140009ad0e0, 0xf}, 0x1400000d578, 0x14000164060, {0x140009ad0e0, 0xf}, {0x102b1f083, ...}, ...)
	/Users/geekboy/Github/httpx/runner/runner.go:673 +0xa8
created by github.com/projectdiscovery/httpx/runner.(*Runner).process
	/Users/geekboy/Github/httpx/runner/runner.go:671 +0x624
```

**Comment by Mzack9999** [comment]
Multiple ASNs aren't parsed correctly - Proposed solution at https://github.com/ammario/ipisp/pull/23 (alternatively, forking is an option)

**Review by Mzack9999 — APPROVED** [review]

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**